firestore/src/main/query_main.cc

```diff
@@ -190,10 +210,10 @@ core::Bound QueryInternal::ToBound(
       SimpleThrowInvalidArgument(message);
     }
 
-    components.push_back(std::move(value).value());
+    components->values[i] = *DeepClone(*value).release();
```

Comment:
Question: why is `DeepClone` necessary here?

Reply:
The memory is owned by the DocumentSnapshot and we need to move ownership to the query bound. See here: https://github.com/firebase/firebase-ios-sdk/blob/e8f419baf275899068e915a6c2ecb4ae1e43d264/Firestore/Source/API/FIRQuery.mm#L584